PostHog/PostHogSDK.swift

```diff
@@ -831,6 +862,164 @@ let maxRetryDelay = 30.0
+    ///   - reloadFeatureFlags: Whether to automatically reload feature flags after setting properties
+    @objc(setPersonPropertiesForFlagsWithProperties:reloadFeatureFlags:)
+    public func setPersonPropertiesForFlags(_ properties: [String: Any], reloadFeatureFlags: Bool = true) {
+        if !isEnabled() {
+            return
+        }
+
+        let sanitizedProperties = sanitizeDictionary(properties) ?? [:]
+        guard !sanitizedProperties.isEmpty else { return }
+        remoteConfig?.setPersonPropertiesForFlags(sanitizedProperties)
+
+        if reloadFeatureFlags {
+            remoteConfig?.reloadFeatureFlags()
+        }
+    }
+
+    /// Resets all person properties that were set for feature flag evaluation.
+    ///
+    /// After calling this method, feature flag evaluation will only use server-side person properties
+    /// and will not include any locally overridden properties.
+    ///
+    /// ## Example Usage
+    /// ```swift
+    /// // Clear all locally set person properties for flags
+    /// PostHogSDK.shared.resetPersonPropertiesForFlags()
+    ///
+    /// // Feature flags will now use only server-side properties
+    /// let flagValue = PostHogSDK.shared.isFeatureEnabled("feature")
+    /// ```
+    ///
+    /// - Note: This method does not automatically reload feature flags. Call `reloadFeatureFlags()`
+    ///         after resetting if you want to immediately refresh flags with the cleared properties.
+    @objc public func resetPersonPropertiesForFlags() {
+        if !isEnabled() {
+            return
+        }
+
+        remoteConfig?.resetPersonPropertiesForFlags()
+    }
+
+    /// Sets properties for a specific group type to include when evaluating feature flags.
+    /// These properties supplement the standard group information sent to PostHog for flag evaluation,
+    /// providing additional context that can be used in flag targeting conditions.
+    ///
+    /// ## Example Usage
+    /// ```swift
+    /// PostHogSDK.shared.setGroupPropertiesForFlags("organization", properties: [
+    ///     "plan": "enterprise",
+    ///     "seats": 50,
+    ///     "industry": "technology"
+    /// ])
+    /// ```
+    ///
+    /// - Parameters:
+    ///   - groupType: The group type identifier (e.g., "organization", "team")
+    ///   - properties: Dictionary of properties to set for this group type
+    /// - Note: This method automatically reloads feature flags to apply the new properties.
+    @objc public func setGroupPropertiesForFlags(_ groupType: String, properties: [String: Any]) {
+        if !isEnabled() {
+            return
+        }
+
+        let sanitizedProperties = sanitizeDictionary(properties) ?? [:]
+        guard !sanitizedProperties.isEmpty else { return }
+        remoteConfig?.setGroupPropertiesForFlags(groupType, properties: sanitizedProperties)
+
+        // Automatically reload flags to apply the new properties
+        remoteConfig?.reloadFeatureFlags()
```

Comment:
should we add a method param similar to `reloadFeatureFlags`? so people can disable if needed